src/persist-client/src/impl/state.rs

```diff
@@ -84,6 +84,14 @@ where
             }));
         }
 
+        // Only allow empty batches at empty time intervals.
+        if desc.upper() == desc.lower() && !keys.is_empty() {
+            return Break(Err(InvalidUsage::InvalidBounds {
```

Comment:
hmm InvalidBounds isn't really correct here: the bounds are legal. I think this _might_ be caught by the UpdateNotWithinBounds check in encode_batch, but it's subtle enough that I don't love the idea of depending on that and panic'ing here (especially given that aljsocha is about to make that logic a bit more removed so that we can have better retry behavior). let's add a new entry to InvalidUsage for this case

Reply:
what would you want to name that variant? thats what stumped me when i thought about adding a new one :)

Reply:
Went with `InvalidEmptyTimeInterval` but accepting any and all naming suggestions